NDTensors/src/dense/tensoralgebra/contract.jl

```diff
@@ -393,15 +397,20 @@ function _contract!(
 
   #tC = similar(CM)
   #_gemm!(tA, tB, El(α), AM, BM, El(β), CM)
-  @strided mul!(CM, AM, BM, El(α), El(β))
+  mul!!(CM, AM, BM, El(α), El(β))
 
   if props.permuteC
     Cr = reshape(CM, props.newCrange)
     # TODO: use invperm(pC) here?
     #@timeit_debug timer "_contract!: permutedims C" begin
-    @strided CT .= permutedims(Cr, props.PC)
+    #@strided 
+    CT .= permutedims(Cr, props.PC)
```

Comment:
Maybe need to refactor out into `permutedims!!` which then dispatches on `Array` or `CuArray`?